akka-cluster-sharding/src/test/scala/akka/cluster/sharding/LeastShardAllocationStrategySpec.scala

```diff
@@ -18,7 +18,7 @@ class LeastShardAllocationStrategySpec extends AkkaSpec {
   "LeastShardAllocationStrategy" must {
     "allocate to region with least number of shards" in {
       val allocations = Map(regionA -> Vector("shard1"), regionB -> Vector("shard2"), regionC -> Vector.empty)
-      allocationStrategy.allocateShard(regionA, "shard3", allocations) should ===(regionC)
+      allocationStrategy.allocateShard(regionA, "shard3", allocations).value.get.get should ===(regionC)
```

Comment:
Yes, the current ones will work with the above test. It will be just surprising if in the future someone changes the impl to be properly async and then suddenly these test cases fail. Maybe not a big issue, though.